### PR TITLE
Adds Task to Migrate from SS File System to Widget

### PR DESCRIPTION
**Pull request by lbradley-mm**
See updates to `README.md` for the steps of how to use

Possible enhancements:

- [ ] Re-start the migration if the API limit is reached
- [ ] Stop the migration if nearing pre-set API limit (to prevent using entire quota)
- [ ] Make `::extend()` call on class with field name and cloudinary data to allow for additional amends to be made (e.g. alt text/captions/credits)

### CODE REVIEW

File: src/Tasks/MigrationTaskStep2.php [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+        $schema = DataObject::getSchema();
+
+        /**
+         *
+         */
+        foreach($dataObjectClasses As $className) {
+            DataObject::get($className)->each(function (DataObject $do) use ($schema) {
+                $stages = [
+                    Versioned::DRAFT => '',
+                ];
```
Comment (mrai-mm) [inline src/Tasks/MigrationTaskStep2.php:158]: If you want to follow the coding standard defined by PSR-2 or PSR-12 then you should have line break after a line of code.

File: src/Tasks/MigrationTaskStep2.php [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+                if ($do->has_extension(Versioned::class)) {
+                    $stages[Versioned::LIVE] = '_Live';
+                }
```
Comment (mrai-mm) [inline src/Tasks/MigrationTaskStep2.php:161]: Here too.

File: src/Tasks/MigrationTaskStep2.php [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+                '{LinkTable}' => $schema->tableName(ImageLink::class),
+                '{FileTable}' => $schema->tableName(File::class),
+                '{VersionSuffix}' => $tableSuffix,
+                '{RootRelationName}' => $do->ClassName,
+                '{ID}' => $do->ID,
+            ]
+        ));
+
+        $newData = [];
+        foreach($oldData as $item) {
```
Comment (mrai-mm) [inline src/Tasks/MigrationTaskStep2.php:208]: There should be a line break here.

File: src/Tasks/MigrationTaskStep2.php [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+            return $this->cache->get($cacheKey);
+        }
+
+        try {
+            $request = new HTTPRequest('GET', '/', [
+                'public_id' => $publicId,
+                'resource_type' => 'image'
+            ]);
+
+            var_dump(sprintf('Requesting [%d]: %s', ++$this->requestCount, $publicId));
```
Comment (mrai-mm) [inline src/Tasks/MigrationTaskStep2.php:293]: Do we have to use var_dump for these?

**Review by mrai-mm — CHANGES_REQUESTED** [review]
Code looks fine. Just a few coding standard issues. Can you review your code and add line breaks where it needs to be?

File: src/Tasks/MigrationTaskStep2.php (hunk 4 above)
Comment (lbradley-mm) [reply src/Tasks/MigrationTaskStep2.php]: Replaced with `static::output()` (new static function)

File: src/Tasks/MigrationTaskStep2.php (hunk 1 above)
Comment (lbradley-mm) [reply src/Tasks/MigrationTaskStep2.php:154]: Added line break

File: src/Tasks/MigrationTaskStep2.php (hunk 2 above)
Comment (lbradley-mm) [reply src/Tasks/MigrationTaskStep2.php:158]: Added line break

File: src/Tasks/MigrationTaskStep2.php (hunk 3 above)
Comment (lbradley-mm) [reply src/Tasks/MigrationTaskStep2.php:207]: Added line break

**Review by mrai-mm — APPROVED** [review]
👍

**Pull request merged**